### DISCUSSION TITLE
Time dependent ConvectiveHeatFluxBC

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#17443] [Q&A Getting Started] [answered]
This may be a totally obvious question, but I would like to make the T_infinity value of an ADConvectiveHeatFluxBC or ConvectiveHeatFluxBC depend upon time, or at different timesteps different values of the T_infinity would be changeable. I've been able to have a functional description of the HTC, but not the T_infinity. Any suggestions?

**Comment by GiudGiud** [comment, marked as answer]
T_infinity is a material property [ADConvectiveHeatFluxBC](https://mooseframework.inl.gov/source/bcs/ADConvectiveHeatFluxBC.html), same as the heat transfer coefficient, so it can be set using a [GenericFunctionMaterial](https://mooseframework.inl.gov/source/materials/GenericFunctionMaterial.html), which will use a function with the desired time dependence of the Tinf

Guillaume
